File: 3016. Minimum Number of Pushes to Type Word II/Solution.java
class Solution {
    public int minimumPushes(String word) {
        int[] frequency = new int[26];
        for (int i = 0; i < word.length(); i++) {
            frequency[word.charAt(i) - 'a']++;
        }
        Arrays.sort(frequency);
        int i = 25;
        int count = 0;
        int multiplier = 1;
        int result = 0;
        while (i >= 0 && frequency[i] != 0) {
            result += (multiplier * frequency[i]);
            count++;
            if (count == 8) {
                multiplier++;
                count = 0;
            }
            i--;
        }
        return result;
    }
}

File: 3016. Minimum Number of Pushes to Type Word II/solution.py
class Solution(object):
    def minimumPushes(self, word):
        # Initialize a frequency array
        frequency = [0] * 26

        # Count character frequencies
        for char in word:
            frequency[ord(char) - ord('a')] += 1

        # Sort the frequency array
        frequency.sort()

        # Initialize variables for key press assignment
        i = 25
        count = 0
        multiplier = 1
        result = 0

        # Assign key presses based on frequency
        while i >= 0 and frequency[i] != 0:
            result += (multiplier * frequency[i])
            count += 1
            if count == 8:
                multiplier += 1
                count = 0
            i -= 1

        # Return the total key presses
        return result
